cmd/soroban-cli/src/commands/keys/fund.rs

```diff
@@ -22,13 +53,174 @@ pub struct Cmd {
+        if network.network_passphrase == LOCAL_NETWORK_PASSPHRASE {
+            let cmd = commands::contract::deploy::asset::Cmd::parse_arg_vec(&[
+                "--asset",
+                "native",
+                "--source-account",
+                &self.address.name,
+                "--rpc-url",
+                rpc_url,
+                "--network-passphrase",
+                &network.network_passphrase,
+            ])?;
+            match cmd.run().await {
+                Ok(()) => {
+                    let contract_id =
+                        contract_id_hash_from_asset(&asset, &network.network_passphrase)?;
+                    Ok(stellar_strkey::Contract(contract_id.0))
+                }
+                Err(err) => {
+                    if err.to_string().contains("ExistingValue") {
+                        let contract_id =
+                            contract_id_hash_from_asset(&asset, &network.network_passphrase)?;
+                        Ok(stellar_strkey::Contract(contract_id.0))
+                    } else {
+                        Err(Error::AssetDeploymentError(err.to_string()))
+                    }
+                }
+            }
+        } else {
+            let contract_id = contract_id_hash_from_asset(&asset, &network.network_passphrase)?;
+            Ok(stellar_strkey::Contract(contract_id.0))
+        }
+    }
+
+    pub async fn add_funds(&self, amount: i64) -> Result<(), Error> {
+        let id = self.get_asset_contract_id().await?;
+        let addr = self.address.public_key()?;
+        let rpc_url = self.network.rpc_url.as_ref().ok_or(Error::MissingRpcUrl)?;
+        let network_passphrase = self
+            .network
+            .network_passphrase
+            .as_ref()
+            .ok_or(Error::MissingNetworkPassphrase)?;
+        let temp_account = self.create_temp_account().await?;
+        let temp_secret = format!("{}", temp_account.1);
+        let from_id = format!("{}", temp_account.0);
+        let to_id = format!("{addr}");
+        let cmd = commands::contract::invoke::Cmd::parse_arg_vec(&[
+            "--id",
+            &id.to_string(),
+            "--source-account",
+            &temp_secret,
+            "--rpc-url",
+            rpc_url,
+            "--network-passphrase",
+            network_passphrase,
+            "--",
+            "transfer",
+            "--to",
+            &to_id,
+            "--from",
+            &from_id,
+            "--amount",
+            &amount.to_string(),
+        ])?;
+        match cmd
+            .run(&global::Args {
+                locator: self.address.locator.clone(),
+                filter_logs: Vec::default(),
+                quiet: false,
+                verbose: false,
+                very_verbose: false,
+                list: false,
+                no_cache: false,
+            })
+            .await
+        {
+            Ok(()) => Ok(()),
+            Err(e) => Err(Error::FundTransferError(e.to_string())),
```

Comment:
You should use `.map_err` instead of the match. If you make the inner value the `invoke::Error` type you can do.

```
.map_err(Error::FundTransferError)
```

Reply:
I couldn't map directly to FundTransferError because if I accept the Invoke Error type it created a recursive dependency. Write now  I map it to a string. I think I could also wrap it in a Box if you'd rather go that route.